tests/verify.rs

```diff
@@ -0,0 +1,41 @@
+const CRATE: &str = "test-flip-link-app";
+
+#[test]
+fn should_link_example_firmware() -> anyhow::Result<()> {
+    // Arrange
+    cargo::install_flip_link();
+
+    // Act
+    cargo::build_example_firmware()
+
+    // Assert
```

Comment:
is this intentionally left blank?

Reply:
Yes. For tests I generally like to separate it into the three stages "Arrange", "Act", "Assert". But in this case I don't really know what to assert, but thought that I will just keep it in there.

In theory we could check if the binaries exist? But the `cargo build` should fail, so I am not sure if would be useful.

Reply:
What do you think?

Reply:
The reason I asked is because I missed the `?` and thought "well this test will never fail, won't it" 😅. I really like the partition you've introduced– in this case it'd be helpful for future readers to leave a small comment like `// if build_example_firmware() hasn't failed, this test was successful` or, more explicitly, swap the `?` for something like 
```rust
let example_firmware = cargo::build_example_firmware(CRATE);

assert!(example_firmware.is_ok())
```

Reply:
> ```rust
> let example_firmware = cargo::build_example_firmware(CRATE);
> 
> // Assert
> assert!(example_firmware.is_ok())
> ```

That's a nice idea, but sadly doesn't work, since the command invocation in this case doesn't return an error, but panics if it fails. Let me see if we can change this.

Reply:
What do you think? https://github.com/knurling-rs/flip-link/blob/83fda8bb24cacdcb601c45796edcd0ec68467a97/tests/verify.rs#L8-L12 (`.success` panics if the command fails)

Reply:
yes, much more obvious!